### PR TITLE
Add files affected by a commit to gitea API -- similar to github

### PR DESCRIPTION
**Pull request by jasugun**
Hi,

The idea is to have ou cis tool track  files affected from polled changes at compile time for our gitea projects - as currently possible with github projects.
The current API returns a whole commit tree instead of a simple list of affected files. We'd like to implement this list, just like in github.

Thanks,
Laurent Cahour

### CODE REVIEW

**Comment by lafriks** [comment]
Please `make fmt` to fix lint problems

**Comment by lafriks** [comment]
Also you need to update swagger definition

**Comment by techknowlogick** [comment]
For any reviewer looking for GH docs, they can be found here: https://docs.github.com/en/rest/reference/repos#get-a-commit

**Comment by lafriks** [comment]
But if I understand correctly github returns list of objects not strings

File: modules/structs/repo_commit.go [hunk 1]
```diff
@@ -45,6 +45,7 @@ type Commit struct {
 	Author     *User         `json:"author"`
 	Committer  *User         `json:"committer"`
 	Parents    []*CommitMeta `json:"parents"`
+	Files      []string      `json:"files"`
```
Comment (6543) [inline modules/structs/repo_commit.go]: this is different to GH api responce so either rename it ... or make it similar

- sorry

**Review by 6543 — CHANGES_REQUESTED** [review]

File: modules/structs/repo_commit.go (hunk 1 above)
Comment (jasugun) [reply modules/structs/repo_commit.go]: Hi there, sorry for the delay.
I'll fix the errors mentioned above.
Also, I'll look back at how github handle things : I based my code upon our existing handling of github api, might be outdated. 
Thanks for the review anyway.
Laurent

**Comment by jasugun** [comment]
Hi there, I've psuhed a new iteration:

- Fix missing swagger template (forgot to add in the first place)
- Did go fmt the pushed files.
- Gone through the github doc and now use a response the resembles github.
>  I just a list of filenames for now. We can always add the remaining stuff later.
>  adds the following to my test response : "files":[{"filename":"README.md"}]

Thanks for your time.
Laurent

**Comment by jasugun** [comment]
Still not good for the lint sorry. Let me fix that.

**Comment by jasugun** [comment]
Hi there, 

To be honest, I'm not sure what's the issue with the build-backend-no-gcc failure.
gitea_no_gcc builds just fine from this branch.
Coudl anyone advise on what to do please?

Thanks,
Laurent

**Comment by jasugun** [comment]
Hi there,

Can your guys please take a look and tell me if this is acceptable or not?

Thanks a lot for your time,
Laurent

**Comment by 6543** [comment]
now it will **not** colide and can go in as is :)

only one thing: we need tests to make sure it wont break in future :)

thanks for the work :+1:

**Comment by jasugun** [comment]
Thanks a lot for the quick response!

**Comment by 6543** [comment]
@jasugun do you need help writting tests?

**Comment by 6543** [comment]
@jasugun can you resolve the conflict?

**Comment by zeripath** [comment]
... or allow maintainers to update the PR to do it for you

**Comment by 6543** [comment]
> ... or allow maintainers to update the PR to do it for you

not possible due to forked to an org

**Comment by 6543** [comment]
followup: #14579

**Pull request closed without merging**